README.md

```diff
@@ -108,6 +108,73 @@ HTML styled "just right".
 Jekyll will print a local URL where the site can be viewed (usually,
 [http://0.0.0.0:4000/](http://0.0.0.0:4000/)).
 
+### Testing using Docker environment 
+
+#### Build environment
+A containerized development environment can be built using the local
+Dockerfile.
+
+Run the build-images.sh script to generate the development environment and
+associated images.
+
+```bash
+./_scripts/build-images.sh
+```
+
+This action will produce two containers: `webdev` and `webdev-validator`.
+The webdev container will execute a `jekyll serve` command with the
+polling option enabled.
+
+This provides the ability to immediately review rendered content changes.
+
+```bash
+docker run -d -v "$PWD":/site -p 4000:4000 webdev
+```
+
+Shell access can be obtained by overriding the default container command.
+
+This is useful for adding new gems, or modifying the Gemfile.lock for updating
+existing dependencies.
+
+```bash
+docker run -v "$PWD":/site -it webdev /bin/bash
+```
+
+Mounting the local directory as a volume is recommended to ensure that Gemfile and
+Gemfile.lock stay updated with any dependency changes.
```

Comment:
Is there a simple command to do that mounting that we can add to this README, or is that what is being done by the `-v` command flag? In trying to figure this out, I saw https://docs.docker.com/storage/bind-mounts/, which recommends using `--mount` instead.

Reply:
Yes the mounting is being done by the `-v` option. 
Line 131 should be the exact command someone needs to run to render their changes locally with the container. 

Mount is a bit more verbose with the only requirement that the target directory must exist prior to the mount operation while `-v` has no such requirement. Since we control the creation of the container, we would just need to create the target directories as part of our build steps. 

 If we swapped the commands out, the new one would look like this: 
` docker run -d --mount type=bind,source="$pwd",target=/site -p 4000:4000 webdev`
vs `docker run -d -v "$pwd":/site -p 4000:4000 webdev`

Personally I prefer the shorthand command, but I'll defer if  using `--mount` helps relate things together.

Reply:
Given the verbosity of `--mount`, and the ubiquity of `-v`, I'm inclined to agree `-v` is better. But, I think the wording in the README should make it more clear what's happening. Instead of saying `Mounting... is recommended`, which sounds like we're recommending the user *do something extra*... we say something like:

```
The `-v` flag mounts the directory to ensure that updates to the directory inside
the container are reflected in your local environment so you can create a commit
and submit a PR.
```

Reply:
Ok, I've rewritten the README to include more direct wording regarding mounting the local directory for both the shell access command, and the original webdev run command. 

Let me know what you think!

Reply:
Now that I'm understanding the Docker stuff better, I think we can streamline the container builds by just having one container (see earlier comment). I'll re-review the doc updates after that stuff is resolved. I actually still think the validation stuff should be done in a separate PR entirely. That would streamline this PR and we could get this merged in right away, doc updates included. With the validator stuff still up in the air (two containers vs. one container), I'm not sure what changes might still need to be made in the docs.